captioning/detect_caption.py

```diff
@@ -0,0 +1,46 @@
+from .utils import get_detectron_features
+from pythia.common.sample import Sample, SampleList
+import gc
+import torch
+
+# Wrapper class over Detection and Captioning model that defines the
+# combined the caption generation from raw image path
+class DetectCaption:
+
+    def __init__(self, detection_model, caption_model, caption_processor, text_processor, cuda_device):
+        self.detection_model = detection_model
+        self.caption_model = caption_model
+        self.cuda_device = cuda_device
+        self.caption_processor = caption_processor
+        self.text_processor = text_processor
+
+    def predict(self, url, feat_name, get_features=False):
+        with torch.no_grad():
+            detectron_features = get_detectron_features(
+                [url],
+                self.detection_model,
+                False,
+                feat_name,
+                self.cuda_device
+            )
+            # returns a single-element list
+            detectron_features = detectron_features[0]
+
+            sample = Sample()
+            sample.dataset_name = "coco"
+            sample.dataset_type = "test"
+            sample.image_feature_0 = detectron_features
+            sample.answers = torch.zeros((5, 10), dtype=torch.long)
```

Comment:
what's this `(5, 10)` for?

Reply:
Also not clear why anything to do with `answers` has to be inside this captioning wrapper.

Reply:
Yeah, it makes me uncomfortable too :disappointed: , this is coming from the colab demo we used as reference: 
https://colab.research.google.com/drive/1vzrxDYB0vxtuUy8KCaGxm--nDCJvyBSg#scrollTo=UCD0nso8YelA&forceEdit=true&offline=true&sandboxMode=true

After digging inside the forward pass of captioning model `sample.answers` is used on this line: 
https://github.com/facebookresearch/pythia/blob/master/pythia/models/butd.py#L113

Reply:
After digging inside the forward pass of captioning model `sample.answers` is used on this line: 
https://github.com/facebookresearch/pythia/blob/master/pythia/models/butd.py#L113